src/Data/Array.purs

```diff
@@ -436,19 +435,19 @@ foreign import sortImpl :: forall a. (a -> a -> Int) -> Array a -> Array a
 --------------------------------------------------------------------------------
 
 -- | Extract a subarray by a start and end index.
-foreign import slice :: forall a. Int -> Int -> Array a -> Array a
+foreign import slice :: Int -> Int -> Array ~> Array
 
 -- | Keep only a number of elements from the start of an array, creating a new
 -- | array.
-foreign import take :: forall a. Int -> Array a -> Array a
+foreign import take :: Int -> Array ~> Array
 
 -- | Calculate the longest initial subarray for which all element satisfy the
 -- | specified predicate, creating a new array.
 takeWhile :: forall a. (a -> Boolean) -> Array a -> Array a
 takeWhile p xs = (span p xs).init
```

Comment:
Unrelated, but I just noticed this involves allocating two arrays when we only return one (`init`).